generate_makefile.bash

```diff
@@ -291,6 +291,9 @@ do
     --with-sycl)
       update_kokkos_devices Sycl
       ;;
+    --with-threads)
+      update_kokkos_devices Threads
+      ;;
     --with-pthread)
```

Comment:
Should we add a deprecation warning/message here? Might be worth getting it at generation time rather than only when make is run.

Reply:
Done